### DISCUSSION TITLE
Hello, Layout question for you: How can the text editor widget be globally accessible to it?

### DISCUSSION THREAD

**Discussion by ddkwork** [cogentcore/core#923] [Q&A] [answered]
see png
<img width="959" alt="{9CF17253-F701-4162-BB65-B0A009324CF9}" src="https://github.com/cogentcore/core/assets/19886504/040f1083-552a-4ace-9066-8b757a96acec">

**Comment by ddkwork** [comment]
I suddenly thought of a thought: I should move all the button event code to the back of the log display widget, at this time, although the log widget is not the most forward, but relative to the button event code it is still in the front, it should be accessed. However, this idea will break the normal business logic, and after a long time, I may not understand why I wrote it.

**Reply by kkoreilly** [reply, marked as answer]
If you are trying to access a widget in an event handler before you make it, you should do something like this:

```go
var logView *texteditor.Editor

gi.NewButton(parent).SetText("Button").OnClick(func(e events.Event) {
     logView.SomeFunction()
})

// later...

logView = texteditor.NewEditor()
```

**Reply by ddkwork** [reply]
Nice work, I was worried about the null pointer problem with this, but logically it's safe because the global variable is already populated with pointers that have values when the click event occurs, by the time the layout is populated. Thanks again, it's much easier to layout with this idea!
